### PR TITLE
Embed placement.Storage in placement.Service

### PR DESCRIPTION
**Pull request by cw9**
placement.Service used to expose Delete(), SetPlacement and Placement(), which are also exposed from placement.Storage interface, now that we also need to expose SetProto() and Proto(), this pr made placement.Storage embedded in placement.Service interface so that future updates in the placement.Storage interface are available through placement.Service as well.

this pr also moved Dryrun logic to placement.Storage.

@xichen2020 @prateek @jeromefroe

### CODE REVIEW

File: placement/service/service.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ import (
 )
 
 type placementService struct {
-	s        placement.Storage
+	placement.Storage
 	opts     placement.Options
```
Comment (xichen2020) [inline placement/service/service.go]: nit: blank line after embedded type

File: placement/service/service.go [hunk 2]
```diff
@@ -110,18 +105,13 @@ func (ps placementService) AddReplica() (placement.Placement, error) {
 		return nil, err
 	}
 
-	if ps.opts.Dryrun() {
-		ps.logger.Info("this is a dryrun, the operation is not persisted")
-		return p, err
-	}
-
-	return p, ps.s.CheckAndSet(p, v)
+	return p, ps.Storage.CheckAndSet(p, v)
```
Comment (xichen2020) [inline placement/service/service.go]: nit: just `ps.CheckAndSet(p, v)`

File: placement/service/service.go [hunk 3]
```diff
@@ -147,16 +137,11 @@ func (ps placementService) AddInstances(
 		addingInstances[i] = addingInstance
 	}
 
-	if ps.opts.Dryrun() {
-		ps.logger.Info("this is a dryrun, the operation is not persisted")
-		return p, addingInstances, err
-	}
-
-	return p, addingInstances, ps.s.CheckAndSet(p, v)
+	return p, addingInstances, ps.Storage.CheckAndSet(p, v)
```
Comment (xichen2020) [inline placement/service/service.go]: Same, you don't need to explicitly reference `Storage` to call its methods unless the method is overriden in the placement service.

File: placement/service/service.go [hunk 4]
```diff
@@ -110,18 +105,13 @@ func (ps placementService) AddReplica() (placement.Placement, error) {
...
-	return p, ps.s.CheckAndSet(p, v)
+	return p, ps.Storage.CheckAndSet(p, v)
 }
 
-func (ps placementService) AddInstances(
+func (ps *placementService) AddInstances(
 	candidates []placement.Instance,
 ) (placement.Placement, []placement.Instance, error) {
-	p, v, err := ps.s.Placement()
+	p, v, err := ps.Storage.Placement()
```
Comment (xichen2020) [inline placement/service/service.go]: Same

File: placement/service/service.go [hunk 5]
```diff
@@ -258,31 +233,5 @@ func (ps placementService) MarkInstanceAvailable(instanceID string) error {
 		return err
 	}
 
-	return ps.s.CheckAndSet(p, v)
-}
-
-func (ps placementService) Placement() (placement.Placement, int, error) {
```
Comment (xichen2020) [inline placement/service/service.go]: We no longer need these in the service interface?

File: placement/service/service.go (hunk 1 above)
Comment (cw9) [reply placement/service/service.go]: sg

File: placement/service/service.go (hunk 2 above)
Comment (cw9) [reply placement/service/service.go]: sg

File: placement/service/service.go (hunk 4 above)
Comment (cw9) [reply placement/service/service.go]: sg

File: placement/service/service.go (hunk 3 above)
Comment (cw9) [reply placement/service/service.go]: sg

File: placement/service/service.go (hunk 5 above)
Comment (cw9) [reply placement/service/service.go]: We don't, as it's exposed through the embedded Storage interface

**Pull request merged**